src/sentry/api/endpoints/organization_details.py

```diff
@@ -248,6 +288,55 @@ def validate(self, attrs):
                 )
         return attrs
 
+    def save_trusted_relays(self, incoming, changed_data, organization):
+        timestamp_now = datetime.utcnow().replace(tzinfo=UTC).isoformat()
+        option_key = "sentry:trusted-relays"
+        try:
+            # get what we already have
+            existing = (
+                OrganizationOption.objects.get(organization=organization, key=option_key) or {}
+            )
+            key_dict = {val.get("public_key"): val for val in existing.value}
+        except OrganizationOption.DoesNotExist:
+            key_dict = {}  # we don't have anything set
+            existing = None
+
+        modified = False
+        for option in incoming:
+            public_key = option.get("public_key")
+            existing_info = key_dict.get(public_key, {})
+
+            option["created"] = existing_info.get("created", timestamp_now)
+            option["last_modified"] = existing_info.get("last_modified")
+
+            # check if we modified the current public_key info and update last_modified if we did
+            if (
+                not existing_info
+                or existing_info.get("name") != option.get("name")
+                or existing_info.get("description") != option.get("description")
+            ):
+                option["last_modified"] = timestamp_now
+                modified = True
+
+        if len(incoming) == 0 and len(key_dict) != 0:
+            modified = True  # we have removed all keys
```

Comment:
I'm afraid this is still insufficient. If you had two keys previously and remove one, `modified` will still be `False` but you won't enter this branch. 

I believe that the only way to check this properly is to create two sets of `public_key`s and compare them.

Reply:
you're right ( again)

Reply:
yes, fixed